include/jemalloc/internal/arena_inlines_b.h

```diff
@@ -60,10 +60,22 @@ arena_prof_info_get(tsd_t *tsd, const void *ptr, alloc_ctx_t *alloc_ctx,
 
 	if (unlikely(!is_slab)) {
 		/* extent must have been initialized at this point. */
-		large_prof_info_get(extent, prof_info);
+		large_alloc_prof_info_get(extent, prof_info);
+
+		if (reset && (uintptr_t)prof_info->alloc_tctx > (uintptr_t)1U) {
```

Comment:
So does this mean that we'd need to call `iealloc()` twice?

Reply:
Actually, I feel there should be a better place to call the rest. It’s needed when a sample is removed, right? Then we should group the operations that way.

Reply:
For example, maybe group with prof_tctx_reset?

Reply:
Yes, it would be ideal to call the reset at the time the sampled allocation is removed, but it's not easy given our current code structure. E.g. in the `rallocx()` code path, the original allocation could have been gone in the core `iralloct()` call, and doing prof stuff there may (or may not?) be easy?

This line of reasoning was also the motivation behind #1689.

Reply:
I see what you are saying. Tried following that path for a while, but didn't get anything better... this still feels off to me though. Let's take a step back and rethink how / where we should handle this.